src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -148,7 +135,11 @@ public override async Task<string> Execute(OrchestrationContext innerContext, st
 
                     if (jObj != null)
                     {
-                        await this.HandleOutOfProcExecution(jObj);
+                        bool moreWorkToDo = await this.outOfProcShim.HandleOutOfProcExecutionAsync(jObj);

```

Comment:
I'm not sure I love splitting out the "more work to do" logic out of `HandleOutOfProcExecutionAsync()`. It seems like that logic was within the scope of that method's responsibility, and it puts more onus on the callers of that method to properly handle the case where there is more work to do.

It looks like this was done for testing purposes. Is there way we can still test this without splitting that logic up?